rt_entt_codegen/shared/parallel.cc

```diff
@@ -0,0 +1,233 @@
+#include "parallel.hh"
+
+#include <set>
+
+#include "ecsact/lang-support/lang-cc.hh"
+#include "ecsact/cpp_codegen_plugin_util.hh"
+
+struct loop_iterator_fields {
+	std::vector<ecsact_system_like_id>&                system_list;
+	const std::vector<ecsact_system_like_id>::iterator begin;
+	std::vector<std::vector<ecsact_system_like_id>>&   parallel_system_cluster;
+};
+
+auto loop_iterator(
+	std::vector<ecsact_system_like_id>&                system_list,
+	const std::vector<ecsact_system_like_id>::iterator begin,
+	std::vector<std::vector<ecsact_system_like_id>>&   parallel_system_cluster
+) -> void;
+
+auto ecsact::rt_entt_codegen::parallel::print_parallel_system_executions(
```

Comment:
Since this doesn't only print parallel systems this name seems a tad off to me. 
What I think might be a lot nicer is to create a function that returns the vector you've created and then iterate over those clusters somewhere else instead of having the printing all being done here. It would be a nice separations of concerns.

```suggestion
auto ecsact::rt_entt_codegen::parallel::get_parallel_execution_clusters(...) -> std::vector<std::vector<ecsact_system_like_id>>
```

Then we can have the printing logic back inside `ecsact::rt_entt_codegen::core::print_execute_systems`